reconcile/vault_replication.py

```diff
@@ -0,0 +1,258 @@
+import logging
+import re
+import sys
+
+from reconcile.utils.vault import VaultClient, _VaultClient, SecretNotFound
+from reconcile.gql_definitions.jenkins_configs import jenkins_configs
+from reconcile.gql_definitions.jenkins_configs.jenkins_configs import (
+    JenkinsConfigV1,
+)
+
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.gql_definitions.vault_instances import vault_instances
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultReplicationV1,
+    VaultInstanceV1,
+)
+from reconcile.gql_definitions.vault_policies.vault_policies import (
+    VaultPolicyV1,
+)
+
+from reconcile.utils import gql
+
+from typing import List, cast, Optional
+
+QONTRACT_INTEGRATION = "vault-replication"
+
+
+def deep_copy_versions(
+    dry_run: bool,
+    source_vault,
+    dest_vault,
+    current_dest_version,
+    current_source_version,
+    path,
+):
+    for version in range(current_dest_version + 1, current_source_version + 1):
+        secret_dict = {"path": path, "version": version}
+        copy_secret(
+            dry_run=dry_run,
+            source_vault=source_vault,
+            dest_vault=dest_vault,
+            path=path,
+            secret_dict=secret_dict,
+        )
+
+
+def copy_secret(
+    dry_run: bool,
+    source_vault: _VaultClient,
+    dest_vault: _VaultClient,
+    path: str,
+    secret_dict: dict,
+) -> None:
+    secret, src_version = source_vault.read_all_with_version(secret_dict)
+    write_dict = {"path": path, "data": secret}
+    logging.info(["replicate_vault_secret", src_version, path])
```

Comment:
how is this going to work with `v1` secrets that return `src_version == None`

Reply:
But there's no version in v1 secrets.

Reply:
Fixed, I was not understanding the comment, sorry!